File: solution/data_structure/10866/main.cpp
// Authored by : tony9402
// Co-authored by : -
// Link : http://boj.kr/29a0795cb69c4aefa644a07f7019bbb9
#include<bits/stdc++.h>

using namespace std;

int main(){
    ios::sync_with_stdio(false);
    cin.tie(0);

    int N; cin >> N;
    deque<int> dq;
    for(int i=0;i<N;i++) {
        string cmd; cin >> cmd;
        if(cmd == "push_front") {
            int X; cin >> X;
            dq.push_front(X);
        }
        else if(cmd == "push_back") {
            int X; cin >> X;
            dq.push_back(X);
        }
        else if(cmd == "pop_front") {
            if(dq.empty()) cout << -1 << '\n';
            else {
                cout << dq.front() << '\n';
                dq.pop_front();
            }
        }
        else if(cmd == "pop_back") {
            if(dq.empty()) cout << -1 << '\n';
            else {
                cout << dq.back() << '\n';
                dq.pop_back();
            }
        }
        else if(cmd == "size") {
            cout << (int)dq.size() << '\n';
        }
        else if(cmd == "empty") {
            cout << dq.empty() << '\n';
        }
        else if(cmd == "front") {
            if(dq.empty()) cout << -1 << '\n';
            else cout << dq.front() << '\n';
        }
        else if(cmd == "back") {
            if(dq.empty()) cout << -1 << '\n';
            else cout << dq.back() << '\n';
        }
    }

    return 0;
}

File: solution/data_structure/10866/main.py
# Authored by : gusdn3477
# Co-authored by : -
# Link : http://boj.kr/1cf1ea352dba44daa9768b67c7f109e9
import sys
from collections import deque

def input():
    return sys.stdin.readline().rstrip()

queue = deque()
N = int(input())
for i in range(N):
    command = input().split()
    if command[0] == 'push_front':
        queue.appendleft(command[1])
    elif command[0] == 'push_back':
        queue.append(command[1])
    elif command[0] == 'pop_front':
        if queue:
            print(queue.popleft())
        else:
            print(-1)
    elif command[0] == 'pop_back':
        if queue:
            print(queue.pop())
        else:
            print(-1)
    elif command[0] == 'size':
        print(len(queue))
    elif command[0] == 'empty':
        if not queue:
            print(1)
        else:
            print(0)
    elif command[0] == 'front':
        if queue:
            print(queue[0])
        else:
            print(-1)
    elif command[0] == 'back':
        if queue:
            print(queue[-1])
        else:
            print(-1)